lib/src/dbs/options.rs

```diff
@@ -77,6 +77,18 @@ impl Options {
 		self.db.as_ref().unwrap()
 	}
 
+	/// Level selected by the user
+	///
+	/// The selected level doesn't imply the user has been authorized to access it.
+	pub fn selected_level(&self) -> Level {
+		match (self.ns.as_ref(), self.db.as_ref()) {
+			(Some(_), Some(_)) => Level::Db,
+			(Some(_), None) => Level::Ns,
+			(None, None) => Level::Kv,
+			_ => unreachable!(),
```

Comment:
Depending on when we call `selected_level` it might currently be possible to trigger the unreachable branch. We currently allow setting namespace and database separately and we don't first check if the namespace is set when setting the database. So `(None, Some(_))` might be reachable depending on how `selected_level` is used.

```bash
$ nix run surrealdb -- sql --endpoint memory
> USE DB database
[]

/database>
```

Reply:
right... I will return an error instead.

Out of curiosity, is this state actually supported? DB defined, NS undefined.

Reply:
actually, I will just return `Level::Db` because the verification check is done somewhere else, and I will refactor this in my next PR anyways